src/activation/useActivateModWizard.test.tsx

```diff
@@ -188,32 +186,35 @@ describe("useActivateModWizard", () => {
     );
     const { result } = renderHook(() => useActivateModWizard(modDefinition));
 
-    expect(result.current.data!.initialValues.optionsArgs).toEqual({
-      [name]: databaseId,
+    await waitFor(() => {
+      expect(result.current.data!.initialValues.optionsArgs).toEqual({
+        [name]: databaseId,
+      });
     });
   });
 
-  test("reject public marketplace activations", () => {
+  test("reject public marketplace activations", async () => {
     const modDefinition = defaultModDefinitionFactory({
       sharing: sharingDefinitionFactory({
         public: true,
       }),
     });
 
-    jest
-      .mocked(useGetFeatureFlagsQuery)
-      .mockReturnValue(
-        valueToAsyncState([
-          mapRestrictedFeatureToFeatureFlag(RestrictedFeatures.MARKETPLACE),
-        ]) as any,
-      );
+    appApiMock.onGet(API_PATHS.FEATURE_FLAGS).reply(200, {
+      flags: [
+        mapRestrictedFeatureToFeatureFlag(RestrictedFeatures.MARKETPLACE),
+      ],
+    });
 
     const { result } = renderHook(() => useActivateModWizard(modDefinition));
 
-    const { isError, error } = result.current;
-    expect(isError).toBe(true);
-    expect(error).toBeInstanceOf(BusinessError);
-    expect((error as BusinessError).message).toBe(
+    let currentResult: ReturnType<typeof useActivateModWizard> | undefined;
+    await waitFor(() => {
+      currentResult = result.current;
+      expect(currentResult.isError).toBe(true);
+    });
+    expect(currentResult?.error).toBeInstanceOf(BusinessError);
+    expect((currentResult?.error as BusinessError).message).toBe(
```

Comment:
NIT: `currentResult` is no shorter than `result.current`, and I'd argue there's actually a slightly greater cognitive lift to it. I'd prefer to get rid of the destructuring and/or reassignment entirely here.

I'd also prefer the `!` operator here so that if for whatever reason `result.current` is undefined, the test failure message will have increased signal (makes it obvious that `result.current` is undefined instead of `result.current.error`)

```suggestion
    await waitFor(() => {
      expect(result.current.isError).toBe(true);
    });
    expect(result.current!.error).toBeInstanceOf(BusinessError);
    expect((result.current!.error as BusinessError).message).toBe(
```